### ISSUE TITLE
Compiling desktop app fails on nodeUtils

### ISSUE THREAD

**Issue by pdarcos** [bitwarden/desktop#484]
I initially opened an issue [here](https://github.com/bitwarden/jslib/issues/126) but it probably makes more sense to post it here since it's related to compiling the desktop app.

Hi,

I was trying to compile the desktop app in ubuntu and when I try npm run electron everything seems to be fine at first until it trips up on nodeUtils

Is this a bug or am I configuring something wrong? The desktop readme doesn't provide much detail. Any help?

ubuntu 18.04.4 LTS
node v12.18.2
npm v6.14.5

Thanks

| ./jslib/src/electron/updater.main.ts 6.34 KiB [built]
| ./jslib/src/electron/window.main.ts 10.8 KiB [built]
| ./jslib/src/services/i18n.service.ts 5.9 KiB [built]
| + 4 hidden modules

ERROR in /home/user/git/bitwarden_desktop/jslib/src/misc/nodeUtils.ts
[tsl] ERROR in /home/user/git/bitwarden_desktop/jslib/src/misc/nodeUtils.ts(20,60)
TS2345: Argument of type 'ReadStream' is not assignable to parameter of type 'ReadLineOptions'.
Property 'input' is missing in type 'ReadStream' but required in type 'ReadLineOptions'.

ERROR in /home/user/git/bitwarden_desktop/jslib/src/misc/nodeUtils.ts
[tsl] ERROR in /home/user/git/bitwarden_desktop/jslib/src/misc/nodeUtils.ts(22,30)
TS7006: Parameter 'line' implicitly has an 'any' type.

ERROR in /home/user/git/bitwarden_desktop/jslib/src/misc/nodeUtils.ts
[tsl] ERROR in /home/user/git/bitwarden_desktop/jslib/src/misc/nodeUtils.ts(26,31)
TS7006: Parameter 'err' implicitly has an 'any' type.

Any ideas?

Thanks

**Comment by kspearrin** [comment]
I am not sure how to reproduce. IT seems to be complaining about typescript errors. Maybe you can try globally installing typescript from npm?

**Comment by pdarcos** [comment]
Thanks for the quick reply @kspearrin 

Unfortunately that didn't work. I ran npm install -g npm and it installed without problems yet every time I run npm run electron I get the error messages above.

Maybe I'm doing something wrong but this is my flow:

1) git clone https://github.com/bitwarden/desktop.git --recursive
2) cd into jslib directory and run npm install
3) cd back into desktop and also run npm install
4) npm run electron
5) I get the errors detailed below

[11] ./src/main.ts + 18 modules 77.8 KiB {0} [built]
     | ./src/main.ts 5.65 KiB [built]
     | ./src/services/i18n.service.ts 961 bytes [built]
     | ./jslib/src/services/constants.service.ts 3.08 KiB [built]
     | ./src/main/menu.main.ts 20.2 KiB [built]
     | ./jslib/src/electron/electronConstants.ts 424 bytes [built]
     | ./src/main/messaging.main.ts 2.97 KiB [built]
     | ./src/main/powerMonitor.main.ts 3.1 KiB [built]
     | ./jslib/src/electron/keytarStorageListener.ts 1.69 KiB [built]
     | ./jslib/src/electron/services/electronLog.service.ts 1.4 KiB [built]
     | ./jslib/src/electron/services/electronMainMessaging.service.ts 444 bytes [built]
     | ./jslib/src/electron/services/electronStorage.service.ts 769 bytes [built]
     | ./jslib/src/electron/tray.main.ts 5.13 KiB [built]
     | ./jslib/src/electron/updater.main.ts 6.34 KiB [built]
     | ./jslib/src/electron/window.main.ts 10.8 KiB [built]
     | ./jslib/src/services/i18n.service.ts 5.9 KiB [built]
     |     + 4 hidden modules

ERROR in /home/user/git/bitwarden_desktop/jslib/src/misc/nodeUtils.ts
[tsl] ERROR in /home/user/git/bitwarden_desktop/jslib/src/misc/nodeUtils.ts(20,60)
      TS2345: Argument of type 'ReadStream' is not assignable to parameter of type 'ReadLineOptions'.
  Property 'input' is missing in type 'ReadStream' but required in type 'ReadLineOptions'.

ERROR in /home/user/git/bitwarden_desktop/jslib/src/misc/nodeUtils.ts
[tsl] ERROR in /home/user/git/bitwarden_desktop/jslib/src/misc/nodeUtils.ts(22,30)
      TS7006: Parameter 'line' implicitly has an 'any' type.

ERROR in /home/user/git/bitwarden_desktop/jslib/src/misc/nodeUtils.ts
[tsl] ERROR in /home/user/git/bitwarden_desktop/jslib/src/misc/nodeUtils.ts(26,31)
      TS7006: Parameter 'err' implicitly has an 'any' type.

ERROR in /home/user/git/bitwarden_desktop/src/app/accounts/hint.component.ts
[tsl] ERROR in /home/user/git/bitwarden_desktop/src/app/accounts/hint.component.ts(17,15)
      TS2345: Argument of type 'import("/home/user/git/bitwarden_desktop/node_modules/@angular/router/src/router").Router' is not assignable to parameter of type 'import("/home/user/git/bitwarden_desktop/jslib/node_modules/@angular/router/src/router").Router'.
  Types have separate declarations of a private property 'rootComponentType'.

ERROR in /home/user/git/bitwarden_desktop/src/app/accounts/lock.component.ts
[tsl] ERROR in /home/user/git/bitwarden_desktop/src/app/accounts/lock.component.ts(26,15)
      TS2345: Argument of type 'import("/home/user/git/bitwarden_desktop/node_modules/@angular/router/src/router").Router' is not assignable to parameter of type 'import("/home/user/git/bitwarden_desktop/jslib/node_modules/@angular/router/src/router").Router'.

ERROR in /home/user/git/bitwarden_desktop/src/app/accounts/login.component.ts
[tsl] ERROR in /home/user/git/bitwarden_desktop/src/app/accounts/login.component.ts(34,28)
      TS2345: Argument of type 'import("/home/user/git/bitwarden_desktop/node_modules/@angular/router/src/router").Router' is not assignable to parameter of type 'import("/home/user/git/bitwarden_desktop/jslib/node_modules/@angular/router/src/router").Router'.

ERROR in /home/user/git/bitwarden_desktop/src/app/accounts/register.component.ts
[tsl] ERROR in /home/user/git/bitwarden_desktop/src/app/accounts/register.component.ts(23,28)
      TS2345: Argument of type 'import("/home/user/git/bitwarden_desktop/node_modules/@angular/router/src/router").Router' is not assignable to parameter of type 'import("/home/user/git/bitwarden_desktop/jslib/node_modules/@angular/router/src/router").Router'.

ERROR in /home/user/git/bitwarden_desktop/src/app/accounts/two-factor-options.component.ts
[tsl] ERROR in /home/user/git/bitwarden_desktop/src/app/accounts/two-factor-options.component.ts(19,28)
      TS2345: Argument of type 'import("/home/user/git/bitwarden_desktop/node_modules/@angular/router/src/router").Router' is not assignable to parameter of type 'import("/home/user/git/bitwarden_desktop/jslib/node_modules/@angular/router/src/router").Router'.

ERROR in /home/user/git/bitwarden_desktop/src/app/accounts/two-factor.component.ts
[tsl] ERROR in /home/user/git/bitwarden_desktop/src/app/accounts/two-factor.component.ts(40,28)
      TS2345: Argument of type 'import("/home/user/git/bitwarden_desktop/node_modules/@angular/router/src/router").Router' is not assignable to parameter of type 'import("/home/user/git/bitwarden_desktop/jslib/node_modules/@angular/router/src/router").Router'.

ERROR in /home/user/git/bitwarden_desktop/src/app/vault/view.component.ts
[tsl] ERROR in /home/user/git/bitwarden_desktop/src/app/vault/view.component.ts(43,55)
      TS2345: Argument of type 'import("/home/user/git/bitwarden_desktop/node_modules/@angular/core/src/zone/ng_zone").NgZone' is not assignable to parameter of type 'import("/home/user/git/bitwarden_desktop/jslib/node_modules/@angular/core/src/zone/ng_zone").NgZone'.
  Types of property 'onUnstable' are incompatible.
    Type 'import("/home/user/git/bitwarden_desktop/node_modules/@angular/core/src/event_emitter").EventEmitter<any>' is not assignable to type 'import("/home/user/git/bitwarden_desktop/jslib/node_modules/@angular/core/src/event_emitter").EventEmitter<any>'.
      Types of property 'subscribe' are incompatible.
        Type '(generatorOrNext?: any, error?: any, complete?: any) => import("/home/user/git/bitwarden_desktop/node_modules/rxjs/internal/Subscription").Subscription' is not assignable to type '(generatorOrNext?: any, error?: any, complete?: any) => import("/home/user/git/bitwarden_desktop/jslib/node_modules/rxjs/internal/Subscription").Subscription'.
          Type 'import("/home/user/git/bitwarden_desktop/node_modules/rxjs/internal/Subscription").Subscription' is not assignable to type 'import("/home/user/git/bitwarden_desktop/jslib/node_modules/rxjs/internal/Subscription").Subscription'.
            Property '_parent' is protected but type 'Subscription' is not a class derived from 'Subscription'.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! bitwarden@0.0.0 build:main: `webpack --config webpack.main.js`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the bitwarden@0.0.0 build:main script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2020-07-16T11_40_43_042Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! bitwarden@0.0.0 electron: `npm run build:main && concurrently -k -n Main,Rend -c yellow,cyan "electron --inspect=5858 ./build --watch" "npm run build:renderer:watch"`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the bitwarden@0.0.0 electron script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2020-07-16T11_40_43_069Z-debug.log

**Comment by kspearrin** [comment]
That is not the process defined in our readme. Simply

```
git clone https://github.com/bitwarden/desktop.git
cd desktop
```

then

```
npm install
npm run electron
```

**Comment by pdarcos** [comment]
Ah that's probably it then.

I had done exactly as the readme indicated before but when I ran npm run electron it had opened a browser window and was apparently stuck there. Then I noticed in the console it was complaining about jslib which is why I manually installed the modules in that directory.
I just started fresh doing exactly as the readme states and indeed it does open after awhile (but I still get the hung window at first).

Thanks!
